1 Lesson/Homework/Garden.rb

```diff
@@ -0,0 +1,145 @@
+class Raspberry
+  @@states = [:Отсутствует, :Цветение, :Зеленая, :Красная]
+
+  def initialize(index)
+    @index = index
+    @state = @@states[0]
+  end
+
+  def grow!
+    if @state != :Красная
+      @state = @@states[(@@states.find_index(@state) + 1) % @@states.size]
+    end
+  end
+
+  def ripe?
+    @state == :Красная
+  end
+
+  def grab!
+    @state = @@states[0]
+  end
+end
+
+class RaspberryBush
+  attr_reader :berries_quantity
+
+  def initialize(berries_quantity)
+    @berries_quantity = berries_quantity
+    @raspberries = berries_quantity.times.map { |i| Raspberry.new(i) }
+  end
+
+  def grow_all!
+    @raspberries.each { |berry| berry.grow! }
```

Comment:
```suggestion
    @raspberries.each(&:grow!)
```
Or better to use `map`

Reply:
```suggestion
    @raspberries.map(&:grow!)
```